### PR TITLE
[#177001436] Info from PM

### PR DESCRIPTION
**Pull request by vito80ba**
This PR implements the call to PM rendering detailed info about payment instrument.

### CODE REVIEW

File: src/components/PaymentMethod.tsx [hunk 1]
```diff
@@ -102,12 +105,37 @@ export const PaymentMethod: React.FunctionComponent<PaymentMethodProps> = props
...
+          )}
           <div className="col-12 py-2"></div>
-          <div className="col-sm-2 font-weight-bold">HPAN</div>
+          <div className="col-sm-2 font-weight-bold">
+            HPAN
+            {props.walletItemInfo && (
+              <button
+                className="btn btn-sm btn-primary"
+                onClick={() => {
+                  popModal(props.walletItemInfo);
```
Comment (BurnedMarshal) [inline src/components/PaymentMethod.tsx:128]: We have a typings problem here. `popModal` argument should be `object` and `walletItemInfo` can be undefined. We could be change the signature of this function
```typescript
function popModal(data?: object): void {
    setModalcontent(data === undefined ? "undefined" : JSON.stringify(data, null, 3));
    setModalstate(!modalState);
  }
```

File: src/components/Paymethods.tsx [hunk 2]
```diff
@@ -1,17 +1,40 @@
...
+  }
+
+  useEffect(() => {
+    if (props.wallet) {
+      const walletHash = props.wallet?.data.reduce(
+        (obj, item) => ({ ...obj, [item.hpan]: item }),
+        {}
+      );
+      setWalletInfo(walletHash);
+    }
```
Comment (BurnedMarshal) [inline src/components/Paymethods.tsx]: We can simplify with
```suggestion
   setWalletInfo(
      props.wallet?.data.reduce(
        (obj, item) => ({ ...obj, [item.hpan]: item }),
        {} as Record<string, PublicWalletItem>
      )
    );
```

File: src/components/CitizenData.tsx [hunk 3]
```diff
@@ -19,12 +26,48 @@ export const CitizenData: React.FunctionComponent<CitizenDataProps> = props => {
...
+            apiResponse.mapLeft(err => {
+              return new Error(readableReport(err));
+            })
+          )
+      )
+      .mapLeft(_ => {
+        // TODO: Validation Error
+      })
+      .map(_ => {
+        if (_.status === 200) {
```
Comment (BurnedMarshal) [inline src/components/CitizenData.tsx:62]: Should we handle the other status code of this API?
- 404 User not found into the PM
- 429 Rate limit exceeded
- 500 Internal server error

**Review by BurnedMarshal — CHANGES_REQUESTED** [review]

File: src/components/CitizenData.tsx [hunk 4]
```diff
@@ -19,12 +26,58 @@ export const CitizenData: React.FunctionComponent<CitizenDataProps> = props => {
   const { t } = useTranslation();
   const [modalState, setModalstate] = useState<boolean>(false);
   const [modalContent, setModalcontent] = useState<string>("");
+  const [resultDataWallet, setResultdataWallet] = useState<Wallet | undefined>(
+    undefined
+  );
+  const [resultErrWallet, setResulterrWalletWallet] = useState<string>("");
```
Comment (BurnedMarshal) [inline src/components/CitizenData.tsx]: This should be
```suggestion
  const [resultErrWallet, setResulterrWallet] = useState<string>("");
```

**Review by BurnedMarshal — CHANGES_REQUESTED** [review]

File: src/components/CitizenData.tsx (hunk 4 above)
Comment (vito80ba) [reply src/components/CitizenData.tsx]: you're right

**Review by BurnedMarshal — APPROVED** [review]
lgtm

File: src/components/OrphanPaymethods.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+export const OrphanPaymethods: React.FunctionComponent<PaymethodsProps> = ({
+  paylistItems,
+  walletItems
+}) => {
+  const { t } = useTranslation();
+  const [modalState, setModalstate] = useState<boolean>(false);
+  const [modalContent, setModalcontent] = useState<string>("");
+  const paylistHPANs = Object.keys(paylistItems);
+  const walletHPANs = Object.keys(walletItems);
+  const orphans = walletHPANs.filter(x => !paylistHPANs.includes(x));
```
Comment (BurnedMarshal) [inline src/components/OrphanPaymethods.tsx]:
```suggestion
  const onlyWalletHPANs = walletHPANs.filter(x => !paylistHPANs.includes(x));
```

File: src/i18n.ts [hunk 6]
```diff
@@ -9,7 +9,7 @@ const resources = {
       "Find citizen": "Find citizen",
       Submit: "Submit",
       "Citizen profile": "Citizen profile",
-      "Payment methods": "Payment methods",
+      "Payment methods": "Payment methods Cashback enabled",
```
Comment (BurnedMarshal) [inline src/i18n.ts]: Could be enabled or not.
```suggestion
      "Payment methods": "Payment methods Cashback",
```

**Review by BurnedMarshal — CHANGES_REQUESTED** [review]

**Review by BurnedMarshal — APPROVED** [review]

**Pull request merged**